configure.ac

```diff
@@ -70,6 +70,9 @@ esac
 CFLAGS="-W $CFLAGS"
 
 warn_CFLAGS="-std=c89 -pedantic -Wall -Wextra -Wcast-align -Wnested-externs -Wshadow -Wstrict-prototypes -Wundef -Wno-unused-function -Wno-long-long -Wno-overlength-strings"
+if test x"$CC" = x"clang"; then
+    warn_CFLAGS="$warn_CFLAGS -Wconditional-uninitialized"
+fi
```

Comment:
This is somewhat coarse: It won't match cases like `CC=clang-9` and it could match old clang versions which don't support the flag.

The proper way is to check support for the flag directly. Two possible approaches can be found in some of our PRs:

 * simple direct way: https://github.com/bitcoin-core/secp256k1/pull/833/files#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810 
 * using autoconf "libraries": see #744 

In this case, I'd prefer the simple direct way (but I may be biased here because it's my PR).

Alternatively, feel free to drop this commit and not add the warning at all.